openstack_plugin/compact.py

```diff
@@ -0,0 +1,154 @@
+# #######
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Local imports
+from openstack_plugin.constants import USE_EXTERNAL_RESOURCE_PROPERTY
+
+
+FLAVOR_RESOURCE_CONFIG = (
+    'name',
+    'ram',
+    'disk',
+    'vcpus'
+)
+
+AGGREGATE_RESOURCE_CONFIG = (
+    'name',
+    'availability_zone'
+)
+
+IMAGE_RESOURCE_CONFIG = (
+    'name',
+    'container_format',
+    'disk_format',
+    'tags'
+)
+
+KEYPAIR_RESOURCE_CONFIG = (
+    'name',
+    'public_key',
+)
+
+
+class Compact(object):
+    def __init__(self, context):
+        """
+        This will set current node context in order to help do the
+        transformation process
+        :param context: Cloudify context cloudify.context.CloudifyContext
+        """
+        self.context = context
+        self._properties = self.context.node.properties
+        self._type = self.context.node.type
+
+    def get_transformation_handler(self):
+        return {
+            'cloudify.openstack.nodes.Flavor': self._transform_flavor,
+            'cloudify.openstack.nodes.HostAggregate':
+                self._transform_aggregate,
+            'cloudify.openstack.nodes.Image': self._transform_image,
+            'cloudify.openstack.nodes.KeyPair': self._transform_keypair
+        }
+
+    def get_common_properties(self):
+        """
+        This method will prepare common compatible openstack version 3
+        properties
+        :return dict: Common Compatible openstack version 3 properties
+        """
+        common = dict()
+        # Populate client config from openstack config
+        if self._properties.get('openstack_config'):
+            client_config = dict()
+            # Parse openstack config so that we can populate the client
+            # config for the `client_config` object
+            for key, value in self._properties['openstack_config'].items():
+                if key == 'tenant_name':
+                    key = 'project_name'
```

Comment:
I don't think we should start changing project and tenant because they are two different things.

Reply:
Actually this is just a mapping between the two plugins since one use `tenant_name` and the other use `project_name`